### PR TITLE
Packaging ag-grid 17.0.0

### PR DESCRIPTION
**Pull request by oliyh**
Thanks

### CODE REVIEW

**Comment by oliyh** [comment]
I'm not 100% sure why the build failed as running `boot package install` worked locally - the error in CircleCI is:

```
Command failed: autoreconf -fiv
/bin/sh: 1: autoreconf: not found

    at ChildProcess.exithandler (child_process.js:275:12)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:567:12)

> optipng-bin@3.1.4 postinstall /home/circleci/.boot/cache/tmp/home/circleci/cljsjs/ag-grid/567/-i5jdxr/ag-grid-17.0.0/node_modules/optipng-bin
> node lib/install.js

  ✔ optipng pre-build test passed successfully

> pngquant-bin@4.0.0 postinstall /home/circleci/.boot/cache/tmp/home/circleci/cljsjs/ag-grid/567/-i5jdxr/ag-grid-17.0.0/node_modules/pngquant-bin
> node lib/install.js

  ⚠ The `/home/circleci/.boot/cache/tmp/home/circleci/cljsjs/ag-grid/567/-i5jdxr/ag-grid-17.0.0/node_modules/pngquant-bin/vendor/pngquant` binary doesn't seem to work correctly
  ⚠ pngquant pre-build test failed
  ℹ compiling from source
  ✔ pngquant pre-build test passed successfully
  ✖ Error: pngquant failed to build, make sure that libpng-dev is installed
    at Promise.all.then.arr (/home/circleci/.boot/cache/tmp/home/circleci/cljsjs/ag-grid/567/-i5jdxr/ag-grid-17.0.0/node_modules/pngquant-bin/node_modules/bin-build/node_modules/execa/index.js:231:11)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pngquant-bin@4.0.0 postinstall: `node lib/install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pngquant-bin@4.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Is this ok or do I need to investigate?

**Comment by oliyh** [comment]
Hi @Deraen I have changed the build to use the dist artifacts as suggested rather than building it from scratch. It now passes CI.

Thanks

**Comment by oliyh** [comment]
Hi I forgot to also update the ag-grid-enterprise and ag-grid-react packages, please hold on and I'll add them to this PR

File: ag-grid-enterprise/build.boot [hunk 1]
```diff
@@ -1,27 +1,17 @@
 (set-env!
   :resource-paths #{"resources"}
-  :dependencies '[[cljsjs/boot-cljsjs "0.9.0" :scope "test"]
-                  [cljsjs/ag-grid "8.0.1-0"]])
```
Comment (Deraen) [inline ag-grid-enterprise/build.boot]: You should keep the ag-grid dependency for the enterprise package.

**Comment by oliyh** [comment]
Hi @Deraen 

Thanks, I am a bit new to this. I saw the dependency declared under the `deps-cljs` directive so thought that took care of it - it's a bit weird to see the previous version declared at the top to me.

Anyway, I put it back and thank you very much for reviewing this so carefully!
Cheers

**Comment by Deraen** [comment]
Deps-cljs require only takes care of ordering the foreign-libs correctly when they are loaded on browser. Maven dependency needs to be added separately so that the using ag-grid-enterprise ensures that ag-grid package is also available (as transitive dependency) if user doesn't use it directly. Deps-cljs is completely separate from Maven dependencies.

**Pull request merged**